raiden/smart_contracts/ChannelManagerLibrary.sol

```diff
@@ -33,6 +33,8 @@ library ChannelManagerLibrary {
     function newChannel(Data storage self, address partner, uint settle_timeout)
         returns (address)
     {
+        require(msg.sender != partner);
```

Comment:
This is not required. It's checked in the `NettingChannelContract` constructor .. and you also put it in the channel library contract above which calls this code.

https://github.com/raiden-network/raiden/blob/e17d96db375d31b134ae7b4e2ad2c1f905b47857/raiden/smart_contracts/NettingChannelContract.sol#L27